protocols/rendezvous/src/codec.rs

```diff
@@ -385,118 +371,110 @@ impl TryFrom<wire::Message> for Message {
-            } => {
-                let registrations = registrations
-                    .into_iter()
-                    .map(|reggo| {
-                        Ok(Registration {
-                            namespace: reggo
-                                .ns
+        let msg_ty = match message.type_ {
+            Some(ty) => ty.enum_value().or(Err(ConversionError::InconsistentWireMessage))?,
+            None => return Err(ConversionError::InconsistentWireMessage),
+        };
+
+        let message = match msg_ty {
+            MessageType::REGISTER => {
+                match message.register.into_option() {
+                    Some(Register{ns, ttl, signedPeerRecord: Some(signed_peer_record), .. }) => {
+                        Message::Register(NewRegistration {
+                            namespace: ns
                                 .map(Namespace::new)
                                 .transpose()?
                                 .ok_or(ConversionError::MissingNamespace)?,
-                            record: PeerRecord::from_signed_envelope(
-                                SignedEnvelope::from_protobuf_encoding(
-                                    &reggo
-                                        .signed_peer_record
-                                        .ok_or(ConversionError::MissingSignedPeerRecord)?,
-                                )?,
-                            )?,
-                            ttl: reggo.ttl.ok_or(ConversionError::MissingTtl)?,
+                            ttl,
+                            record: PeerRecord::from_signed_envelope(SignedEnvelope::from_protobuf_encoding(
+                                &signed_peer_record,
+                            )?)?,
                         })
-                    })
-                    .collect::<Result<Vec<_>, ConversionError>>()?;
-                let cookie = Cookie::from_wire_encoding(cookie)?;
-
-                Message::DiscoverResponse(Ok((registrations, cookie)))
+                    }
+                    _ => return Err(ConversionError::InconsistentWireMessage),
+                }
             }
-            wire::Message {
-                r#type: Some(1),
-                register_response:
-                    Some(RegisterResponse {
-                        status: Some(error_code),
-                        ..
-                    }),
-                ..
-            } => {
-                let error_code = wire::message::ResponseStatus::from_i32(error_code)
-                    .ok_or(ConversionError::BadStatusCode)?
-                    .try_into()?;
-                Message::RegisterResponse(Err(error_code))
+            MessageType::REGISTER_RESPONSE => {
+                match message.registerResponse.into_option() {
+                    Some(
+                        RegisterResponse { status: Some(status), ttl, .. }
+                    ) if status.value() == 0 => Message::RegisterResponse(Ok(ttl.ok_or(ConversionError::MissingTtl)?)),
```

Comment:
This `status` is enum as well right? I think we should probably use that instead of the bare number. With `prost`, we couldn't match on that but now we can :)

Reply:
It's an enum but it's wrapped in a couple of abstractions. Will take a look again.